TestModels/ExternV2/README.md

```diff
@@ -0,0 +1,67 @@
+# SimpleExternV2
+
+This project tests implementing a [dafny extern](https://homepage.cs.uiowa.edu/~tinelli/classes/181/Papers/dafny-reference.pdf#15) using the "V2 Extern" system.
+This models and its shapes are based on to operations in the `Extern` TestModel.
+
+The differences between these two TestModels are:
+* V2 suffixes are appended to classes and operations
+* The NET implementation of the `simple.dafnyexternv2.internaldafny` extern contains an extern method that is not present in other languages (as a demonstration; see below)
+* The `ExternV2Constructor` is moved to `simple.dafnyexternv2.internaldafny.ExternV2Constructor` 
+
+This TestModel also demonstrates a "V2" extern system.
+
+The "V1" extern system applies an `{:extern "..."}` attribute to a module to signal an extern.
+This worked for .NET and Java, where this extern string was valid in both languages.
+However, this does not work for Python and Go, where this string is not applicable.
+
+The "V2" extern system is intended to allow per-language extern strings.
+This system uses the `replaceable` and `replaces` keywords
+in conjunction with the `outer-module` compile flag
+to allow a module to have idiomatic extern names in each target language.
+
+Under this V2 system, a Dafny developer should not directly mark a module as `extern` if the path to the extern module depends on language-specific features.
```

Comment:
Question/Blocking: 
Is it language-specific features? Like only Java can use ACCP.
Or is it package naming? Like only Java should use `software.amazon.cryptography`.

Or, more confusingly, is it both?

Reply:
OK, this question is answered later in this README.
But this sentence is in-accurate, and should be refactored.

A Dafny Developer SHOULD NOT use `extern` if:
1. they will transpile to runtimes that have conflicting package naming requirements
2. the logic will be customized per runtime

For Crypto Tools, (1.) always applies, 
as we transpile to Java and Python, 
which have conflicting package naming requirements.

Reply:
Will update